src/main/kotlin/com/personio/synthetics/builder/browser/step/ScrollBuilder.kt

```diff
@@ -0,0 +1,26 @@
+package com.personio.synthetics.builder.browser.step
+
+class ScrollBuilder {
+    private var x: Int = 0
+    private var y: Int = 0
+
+    fun build(): Pair<Int, Int> {
+        return x to y
+    }
+
+    fun up(value: Int) {
+        y += value
+    }
+
+    fun down(value: Int) {
+        up(-value)
+    }
+
+    fun left(value: Int) {
+        right(-value)
+    }
+
+    fun right(value: Int) {
+        x += value
+    }
```

Comment:
I think using relative positioning might cause hard-to-debug behavior. Because Javascript in the page might affect scroll position independently of what is defined in the synthetic tests.

Lets say for instance:

- Test says `down(10)`
- ... Behind the scenes, JS changes scroll to `+200`
- Test says `up(5)`

Now, what is the `expected` result?

- Should it be `5`, which assumes Synthetic tests overrides the JS behavior
- Should it be `195` which assumes Synthetic tests knows about the current screen state
- Either of them is valid?

I don't know use-cases for pixel-perfect scrolling, but I'd say it might be better to expose only what is supported by DataDog (ie.: absolute scrolling). That way, the final behavior is clear from the code and does not rely on state that might go out-of-sync with time.

What if we rewrite `up(10)` to `scroll { to(15, VERTICAL) }` and `left(10)` to `scroll { to (10, HORIZONTAL) }`

Us `to` makes it clear that absolute position is used.

Reply:
SGTM

Reply:
FYI I've implemented the suggestion above